### DISCUSSION TITLE
Question about what method would be more secure.

### DISCUSSION THREAD

**Discussion by dillfrescott** [samuel-lucas6/Kryptor#41] [Q&A] [answered]
Ok. I'm a bit stumped here. Would encrypting a file using a ~50 MB completely random data keyfile, or encrypting the same file with PGP be more secure?

Any help would be appreciated!

**Comment by dillfrescott** [comment, marked as answer]
Ok reading through the kryptor docs, it would seem as though a keyfile is more secure, especially a keyfile password combo.

**Reply by samuel-lucas6** [reply]
Thank you for checking the docs! I would advocate for using a strong password over a keyfile, but if you want the best security, then a strong password alongside a randomly generated keyfile is best, just riskier given you have to back up the keyfile. There's no need for the keyfile to be that big though; a random keyfile that's 64 bytes is more than enough.

**Reply by dillfrescott** [reply]
Awesome! Thank you for the information! But technically wouldn't a larger keyfile be exponentially more secure? Because instead of guessing 64 bytes correctly, it would be 52 million bytes? I realize this is far overkill for regular computers anyways, but I have no idea about quantum.

**Reply by samuel-lucas6** [reply]
It's important to think about it in the same way as a normal encryption key. Encryption keys are almost always 128-bit or 256-bit, with 256-bit recommended for post-quantum. A 256-bit key is just 32 random bytes. 64 random bytes is equivalent to a 512-bit key, which is already excessive. Few encryption algorithms support a key size larger than 256-bit because it's not necessary. There's no risk of a 256-bit or 512-bit key being bruteforced by quantum computers. Some would even argue a 128-bit key will still be fine.

**Reply by samuel-lucas6** [reply]
I should also say that no matter what the size of the keyfile, it currently gets condensed to a 512-bit value anyway because the file gets hashed.

**Reply by dillfrescott** [reply]
Oh OK. That's good to know! Does Kryptor only read the first 64 bytes of a file anyways? So is a 50 MB file the exact same security wise as a 64 byte key file?

I'm really reaching into the theoretical here because obv nothing will Crack even a 256 bit key in our lifetimes at least...

**Reply by dillfrescott** [reply]
Ohhhh okay I didn't see that last message haha! So it's essentially a 512 bit hash no matter how large the file is!

**Reply by samuel-lucas6** [reply]
Yep, it used to be that only the first 64 bytes were read, but I changed it to hashing. Using a larger file just results in reduced speed hashing the keyfile.
